src/strategies/taraxa-delegation/index.ts

```diff
@@ -0,0 +1,112 @@
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { Contract } from '@ethersproject/contracts';
+import { formatUnits } from '@ethersproject/units';
+import { getAddress } from '@ethersproject/address';
+
+export const author = 'Taraxa-project';
+export const version = '0.1.0';
+
+const abi = [
+  'function getDelegations(address delegator, uint32 batch) view returns (tuple(address account, tuple(uint256 stake, uint256 rewards) delegation)[] delegations, bool end)'
+];
+
+export default interface Delegation {
+  address: string;
+  stake: BigNumber;
+}
+
+interface ContractDelegationInfo {
+  rewards: BigNumber;
+  stake: BigNumber;
+}
+
+export interface ContractDelegation {
+  account: string;
+  delegation: ContractDelegationInfo;
+}
+
+const getTotalDelegationOfAddress = async (
+  address: string,
+  provider: any,
+  blockTag: string | number
+): Promise<Delegation> => {
+  const mainnetDpos = new Contract(
+    '0x00000000000000000000000000000000000000fe',
+    abi,
+    provider
+  );
+
+  let newDelegations: ContractDelegation[] = [];
+  let page = 0;
+  let hasNextPage = true;
+  while (hasNextPage) {
+    try {
+      const allDelegations = await mainnetDpos.getDelegations(address, page, {
+        blockTag
+      });
+
+      newDelegations = [...newDelegations, ...allDelegations.delegations];
+      hasNextPage = !allDelegations.end;
+      page++;
+    } catch (e) {
+      console.error(e);
+      hasNextPage = false;
+    }
+  }
+
+  const finalDelegations = newDelegations.map(
+    (delegation: ContractDelegation) => ({
+      address: delegation.account,
+      stake: delegation.delegation.stake,
+      rewards: delegation.delegation.rewards
+    })
+  );
+  return {
+    address,
+    stake: finalDelegations
+      .map((s) => s.stake)
+      .reduce((a, b) => a.add(b), BigNumber.from(0))
+  };
+};
+
+const getDelegations = async (
+  addresses,
+  provider,
+  snapshot
+): Promise<Record<string, number>> => {
+  const result: Record<string, BigNumberish> = {};
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  for (const address of addresses) {
+    const delegation = await getTotalDelegationOfAddress(
+      address,
+      provider,
+      blockTag
+    );
+    result[address] = delegation.stake;
+  }
+  return Object.fromEntries(
+    Object.entries(result).map(([address, stake]) => [
+      address,
+      parseFloat(formatUnits(stake, 18))
+    ])
+  );
+};
```

Comment:
Hi @VargaElod23 Looks like this will send one request per address? this will not scale :( https://github.com/snapshot-labs/snapshot-strategies#code,Try using muticall instead

Reply:
Thing is that we cannot do that as our `getDelegations(address account, uint batch)` is possibly not returning all delegations as it returns in batches of 100. Would this be a problem @ChaituVR ? 
I was also wondering if we can implement proposal restrictions for creation on token holding. Is that something that is supported in snapshot?

Reply:
Asking the team if there could be some chance this way

Reply:
Hmm, would something like the following be useful:
 - Multicall for all addresses
 - Check for the ones which got `hasMore` as true
 - Multicall only for those addresses again until `hasMore` returns false.

Reply:
@ChaituVR WDYT about he approach above?

Reply:
What is `hasMore` 🙈 It is a method in your contract? It can cause how many extra multicalls?

Reply:
The method that we call returns delegations in a batch of 100 and a bool `returns (Delegation[] delegations, bool hasMore)` . In case there are more than 100 delegations for an address it returns the first 100 and true for the boolean so that you can programmatically loop through to get the remaining ones. Currently the address that has the most delegations on our mainnet has 20-something so we don't expect to have multiple multicalls for a long while. However, in the future it might happen that we need to do multiple multicalls.